cmd/monitor/monitor.go

```diff
@@ -34,6 +34,8 @@ var (
 	zero               = big.NewInt(0)
 	observedPendingTxs historicalRange
 	maxDataPoints      = 1000
+	maxConcurrency     = 10
+	semaphore          = make(chan struct{}, maxConcurrency)
```

Comment:
I think it would be better to create this `semaphore` channel in `func monitor(ctx context.Context) error` and to close it at the end.